distributed/tests/test_worker_state_machine.py

```diff
@@ -1293,3 +1294,35 @@ def test_gather_dep_failure(ws):
     ]
     assert ws.tasks["x"].state == "error"
     assert ws.tasks["y"].state == "waiting"  # Not ready
+
+
+@pytest.mark.parametrize("fail", [True, False])
+def test_executing_cancelled_fetch_executing(ws, fail):
```

Comment:
TODO: This actually fails when using the `ws_with_running_task` fixture because we're not releasing a resource properly. Will try again *after* https://github.com/dask/distributed/pull/6699

Reply:
Could you move this test to test_cancelled_state.py?
I'd rather have tests grouped by *what* they test, not by *how* they test it.
I already added a `-m workerstate` to pytest to run all WorkerState tests throughout the whole test suite (#6706) for this purpose.